Composer/packages/client/src/components/ProjectTree/TriggerCreationModal.tsx

```diff
@@ -261,11 +298,89 @@ export const TriggerCreationModal: React.FC<TriggerCreationModalProps> = (props)
       });
       return;
     }
-    const newDialog = generateNewDialog(dialogs, dialogId, formData, schemas.sdk?.content);
-    if (formData.$kind === intentTypeKey && !isRegEx) {
+
+    if (formData.$kind === intentTypeKey && isLUISnQnA && formData.triggerPhrases) {
+      const newDialog = generateNewDialog(dialogs, dialogId, formData, schemas.sdk?.content, {});
       const newIntent = { Name: formData.intent, Body: formData.triggerPhrases };
       onSubmit(newDialog, newIntent);
+    } else if (formData.$kind === qnaMatcherKey || formData.$kind === onChooseIntentKey) {
+      const lgTemplateId1 = generateDesignerId();
+      const lgTemplateId2 = generateDesignerId();
+      const extraTriggerAttributes = {};
+      if (formData.$kind === onChooseIntentKey) {
+        extraTriggerAttributes['actions[4].prompt'] = `\${TextInput_Prompt_${lgTemplateId1}()}`;
+        extraTriggerAttributes['actions[5].elseActions[0].activity'] = `\${SendActivity_${lgTemplateId2}()}`;
+        const lgTemplates1: LgTemplate[] = [
+          {
+            name: `TextInput_Prompt_${lgTemplateId1}`,
+            body: '[Activity\n\
+    Attachments = ${json(AdaptiveCardJson())}\n\
+]\n',
+          } as LgTemplate,
```

Comment:
Coupled with too much LG logic, makes this module hard to be maintained. I will follow up with you